app/src/main/java/org/mozilla/fenix/settings/account/AccountSettingsFragment.kt

```diff
@@ -213,56 +244,49 @@ class AccountSettingsFragment : PreferenceFragmentCompat() {
                 pref?.let {
                     pref.title = getString(R.string.preferences_sync_now)
                     pref.isEnabled = true
-                    updateLastSyncedTimePref(context!!, pref, failed = true)
+
+                    val failedTime = getLastSynced(requireContext())
+                    accountSettingsStore.dispatch(AccountSettingsAction.SyncFailed(failedTime))
                 }
             }
         }
     }
 
     private val deviceConstellationObserver = object : DeviceConstellationObserver {
         override fun onDevicesUpdate(constellation: ConstellationState) {
-            val deviceNameKey = context!!.getPreferenceKey(R.string.pref_key_sync_device_name)
-            val preferenceDeviceName = findPreference<Preference>(deviceNameKey)
-            preferenceDeviceName?.summary = constellation.currentDevice?.displayName
+            constellation.currentDevice?.displayName?.also {
+                accountSettingsStore.dispatch(AccountSettingsAction.UpdateDeviceName(it))
+            }
         }
     }
 
-    private fun updateSyncingItemsPreference() {
-        val syncCategory = context!!.getPreferenceKey(R.string.preferences_sync_category)
-        val preferencesSyncCategory = findPreference<Preference>(syncCategory) as PreferenceCategory
-        val stringSet = mutableSetOf<String>()
+    private fun updateDeviceName(state: AccountSettingsState) {
+        val deviceNameKey = context!!.getPreferenceKey(R.string.pref_key_sync_device_name)
+        val preferenceDeviceName = findPreference<Preference>(deviceNameKey)
+        preferenceDeviceName?.summary = state.deviceName
```

Comment:
In the fragment, we set both `text` _and_ `summary`. Is this method wrong, or is the other one?

Reply:
I believe both are correct. The only reason why `text` is set in that one area is because it was a fix to #3733.